meson.build

```diff
@@ -1,106 +1,108 @@
 
 if get_option('tracy_no_callstack_inlines')
-  tracy_compile_args += ['-DTRACY_NO_CALLSTACK_INLINES']
+  tracy_common_args += ['-DTRACY_NO_CALLSTACK_INLINES']
 endif
 
 if get_option('tracy_only_localhost')
-  tracy_compile_args += ['-DTRACY_ONLY_LOCALHOST']
+  tracy_common_args += ['-DTRACY_ONLY_LOCALHOST']
 endif
 
 if get_option('tracy_no_broadcast')
-  tracy_compile_args += ['-DTRACY_NO_BROADCAST']
+  tracy_common_args += ['-DTRACY_NO_BROADCAST']
 endif
 
 if get_option('tracy_only_ipv4')
-  tracy_compile_args += ['-DTRACY_ONLY_IPV4']
+  tracy_common_args += ['-DTRACY_ONLY_IPV4']
 endif
 
 if get_option('tracy_no_code_transfer')
-  tracy_compile_args += ['-DTRACY_NO_CODE_TRANSFER']
+  tracy_common_args += ['-DTRACY_NO_CODE_TRANSFER']
 endif
 
 if get_option('tracy_no_context_switch')
-  tracy_compile_args += ['-DTRACY_NO_CONTEXT_SWITCH']
+  tracy_common_args += ['-DTRACY_NO_CONTEXT_SWITCH']
 endif
 
 if get_option('tracy_no_exit')
-  tracy_compile_args += ['-DTRACY_NO_EXIT']
+  tracy_common_args += ['-DTRACY_NO_EXIT']
 endif
 
 if get_option('tracy_no_sampling')
-  tracy_compile_args += ['-DTRACY_NO_SAMPLING']
+  tracy_common_args += ['-DTRACY_NO_SAMPLING']
 endif
 
 if get_option('tracy_no_verify')
-  tracy_compile_args += ['-DTRACY_NO_VERIFY']
+  tracy_common_args += ['-DTRACY_NO_VERIFY']
 endif
 
 if get_option('tracy_no_vsync_capture')
-  tracy_compile_args += ['-DTRACY_NO_VSYNC_CAPTURE']
+  tracy_common_args += ['-DTRACY_NO_VSYNC_CAPTURE']
 endif
 
 if get_option('tracy_no_frame_image')
-  tracy_compile_args += ['-DTRACY_NO_FRAME_IMAGE']
+  tracy_common_args += ['-DTRACY_NO_FRAME_IMAGE']
 endif
 
 if get_option('tracy_no_system_tracing')
-  tracy_compile_args += ['-DTRACY_NO_SYSTEM_TRACING']
+  tracy_common_args += ['-DTRACY_NO_SYSTEM_TRACING']
 endif
 
 if get_option('tracy_patchable_nopsleds')
-  tracy_compile_args += ['-DTRACY_PATCHABLE_NOPSLEDS']
+  tracy_common_args += ['-DTRACY_PATCHABLE_NOPSLEDS']
 endif
 
 if get_option('tracy_delayed_init')
-  tracy_compile_args += ['-DTRACY_DELAYED_INIT']
+  tracy_common_args += ['-DTRACY_DELAYED_INIT']
 endif
 
 if get_option('tracy_manual_lifetime')
-  tracy_compile_args += ['-DTRACY_MANUAL_LIFETIME']
+  tracy_common_args += ['-DTRACY_MANUAL_LIFETIME']
 endif
 
 if get_option('tracy_fibers')
-  tracy_compile_args += ['-DTRACY_FIBERS']
+  tracy_common_args += ['-DTRACY_FIBERS']
 endif
 
 if get_option('tracy_timer_fallback')
-  tracy_compile_args += ['-DTRACY_TIMER_FALLBACK']
-endif
-
-tracy_shared_libs = get_option('tracy_shared_libs')
-if tracy_shared_libs
-  add_project_arguments('-DTRACY_EXPORTS', language : 'cpp')
+  tracy_common_args += ['-DTRACY_TIMER_FALLBACK']
 endif
 
 if get_option('tracy_no_crash_handler')
-  tracy_compile_args += ['-DTRACY_NO_CRASH_HANDLER']
+  tracy_common_args += ['-DTRACY_NO_CRASH_HANDLER']
 endif
 
-add_project_arguments(tracy_compile_args, language : 'cpp')
+tracy_shared_libs = get_option('default_library') == 'shared' or get_option('tracy_shared_libs')
```

Comment:
For future reference: there's also `default_library` = `both`, which will build both libraries at once. However, there is currently no way to set a define only for the shared library in that case. This is an issue about it: https://github.com/mesonbuild/meson/issues/3304

Reply:
since 1.3.0 `library` supports `<lang>_shared_args` and `<lang>_static_args`, as its still in release candidate phase i decided to not use them. Im unsure as to what the recommended way to currently handle this is

Reply:
From what I gathered, the current way everyone is doing it is with an option like `tracy_shared_libs` but yeah idk. I think `default_library` should work just as well? As long as you can override it for subprojects.

Reply:
default_library can be specified per subproject via `-Dsubproject:default_library=xxx` or in the toplevel projects `default_options` field in `project`